### DISCUSSION TITLE
Any plans to control BLE mesh via official WiFi-BLE bridges?

### DISCUSSION THREAD

**Discussion by AMDFreak2006** [fsaris/EspHome-AwoX-BLE-mesh-hub#38] [Q&A]
At least via the AwoX cloud it is possible together with an
- EGLO PLUG **PLUS** (seems to not be in production anymore, but can be bought nonetheless): [click here](https://www.eglo.com/media/catalog/product/dam/TECHDATA/ger-DE/97936.pdf) or
- the new EGLO CONNECT-C Bridge [click here](https://www.eglo.com/de/zubehoer-bridge-c-98576.html#further-information)
to control all AwoX/Eglo BLE-mesh devices also via WiFi (and AwoX cloud).

Not all capabilites of the lights and plugs are available via cloud control:
- plugs are missing power history and automated timer-controlled actions within the device
- lights are missing automatic color animations, automated timer-controlled actions within the device as well as the disco/club function

However the most relevant capabilities are available
- plugs: Switching the state, seeing the currently measured power
- lights: Switching/Dimming (RGB, TW), accessing favorite colours and factory colour themes

Since the BLE control via the EGLO PLUG PLUS is way more stable than the control via BLE directly from the Smartphone (both using the original AwoX app) it might be a good idea to let the official WiFi-BLE bridges handle the whole BLE mesh part and only use the WiFi interface of the bridges directly and/or the AwoX cloud interface.

So my questions:
- Would it be possible to control the BLE mesh
    - directly via AwoX WiFi-BLE bridge(s) using WiFi (offline, no AwoX cloud involved)
    - indirectly via AwoX cloud and an AwoX WiFi-BLE bridge (online, same way as the original AwoX app does it when no BLE connection is available)

**Comment by fsaris** [comment]
The AwoX bridge is what this esphome component tries to bee, but without the cloud requirement, only local control. If you're able to control your HA remotely then you can use that too control the AwoX lights remotely.

Integration with the cloud is out of scope of this component, idea is too be independent of AwoX.

Nevertheless as long as you use the mesh credentials you fetched from the AwoX cloud too control the lights with this component. All actions/state changes should be visible in the cloud too if you have some AwoX device in place that can communicate with the AwoX cloud.
